backend/modules/notifications/service.py

```
@@ -1 +1,335 @@
+        user: User
+    ) -> Dict[str, Any]:
+        """Respond to a cohort invitation (accept or decline)."""
+        logger.info("Responding to invitation %d with action: %s", invitation_id, action)
+        logger.info("Current user ID: %d", user.id)
+        
+        # Find the invitation
+        invitation = self.invitation_repo.get_invitation_by_id(invitation_id)
+        
+        if not invitation:
+            logger.error("Invitation %d not found or not pending", invitation_id)
+            raise ValueError("Invitation not found or already responded to")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)